frontend/src/component/events/EventTimeline/EventTimelineContext.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext } from 'react';
+import type { IEventTimelineContext } from './EventTimelineProvider';
+
+export const EventTimelineContext = createContext<
+    IEventTimelineContext | undefined
+>(undefined);
+
+export const useEventTimelineContext = (): IEventTimelineContext => {
+    const context = useContext(EventTimelineContext);
+
+    if (!context) {
+        throw new Error(
+            'useEventTimelineContext must be used within a EventTimelineProvider',
+        );
```

Comment:
Is this a common pattern? I'm also unsure if `useContext` can return undefined

Reply:
Yes, and it can. See:
 - [Sticky](https://github.com/Unleash/unleash/blob/main/frontend/src/component/common/Sticky/Sticky.tsx#L40)
 - [useFeedback](https://github.com/Unleash/unleash/blob/main/frontend/src/component/feedbackNew/useFeedback.tsx#L23)

The idea is to wrap your component tree with a provider, allowing any nested components to access the context using `useContext`. If a component isn’t within the provider, `useContext` will return undefined. We control the component tree and typically place providers at a high level, so this should never happen. However, if we break it, the UI will loudly throw an error at us.